lms/djangoapps/ccx/overrides.py

```diff
@@ -25,43 +27,42 @@ def get(self, block, name, default):
         """
         Just call the get_override_for_ccx method if there is a ccx
         """
-        ccx = get_current_ccx()
+        # The incoming block might be a CourseKey instance of some type, a
+        # UsageKey instance of some type, or it might be something that has a
+        # location attribute.  That location attribute will be a UsageKey
+        ccx = course_id = None
+        identifier = getattr(block, 'id', None)
+        if isinstance(identifier, CourseKey):
+            course_id = block.id
+        elif isinstance(identifier, UsageKey):
+            course_id = block.id.course_key
+        elif hasattr(block, 'location'):
+            course_id = block.location.course_key
+        else:
+            msg = "Unable to get course id when calculating ccx overide for block type {}"
+            log.error(msg.format(type(block)))
+        if course_id is not None:
+            ccx = get_current_ccx(course_id)
         if ccx:
             return get_override_for_ccx(ccx, block, name, default)
         return default
 
 
-class _CcxContext(threading.local):
+def get_current_ccx(course_id):
     """
-    A threading local used to implement the `with_ccx` context manager, that
-    keeps track of the CCX currently set as the context.
+    Return the ccx that is active for this course.
     """
-    ccx = None
-    request = None
-
-
-_CCX_CONTEXT = _CcxContext()
-
-
-@contextmanager
-def ccx_context(ccx):
-    """
-    A context manager which can be used to explicitly set the CCX that is in
-    play for field overrides.  This mechanism overrides the standard mechanism
-    of looking in the user's session to see if they are enrolled in a CCX and
-    viewing that CCX.
-    """
-    prev = _CCX_CONTEXT.ccx
-    _CCX_CONTEXT.ccx = ccx
-    yield
-    _CCX_CONTEXT.ccx = prev
-
+    # ensure that the ID passed in is a CourseKey instance of some type.
+    if isinstance(course_id, CourseKey):
+        course_key = course_id
+    else:
+        course_key = CourseKey.from_string(course_id)
```

Comment:
Ideally, we'd push all parsing of `course_keys` to the edges of the system, so that the internals are always using `CourseKey` objects, rather than maybe objects, maybe strings.